src/useOverflowDetector.ts

```diff
@@ -25,7 +25,7 @@ export function useOverflowDetector(props: useOverflowDetectorProps) {
   }, [ref.current, props.onChange, setOverflow, overflow]);
 
   useResizeDetector({
-    targetRef: ref as React.MutableRefObject<HTMLElement>,
+    targetRef: ref,
```

Comment:
https://github.com/maslianok/react-resize-detector/pull/234
This change needs react-resize-detector >= 8.0.4.
I'll merge this PR after upgrading dependencies.